micro-rdk/src/common/ota.rs

```diff
@@ -337,6 +342,7 @@ impl<S: OtaMetadataStorage> OtaService<S> {
                 CONN_RETRY_SECS
             );
             Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
+            num_tries += 1;
```

Comment:
- Is there a way to frame this loop without an explicitly managed counter?
- If you really need the counter, I'd recommend incrementing at or near the top so it can't somehow become skipped later.

Reply:
I've tried exploring different paths for a day or two:
- wrapping the connection initialization logic into its own function
  - getting the return type right (even with generics) was a rabbit hole of satisfying the compiler and enabling an unstable/nightly features that are used to define a lower-level trait (something with Allocator if I remember right)
- declaring the variables outside the loop
   -  same issue as above
 - using `for i in 0..CONN_RETRY`
   - can't use `break` in a `for-loop`, only in `loop`

I think I tried a couple other things, (like inspect,map_err, ok_and, etc) but it seemed like too much time spend on trying to avoid the messiness and have the correct behavior.

Reply:
I can move the increment to the top of the loop and set the range accordingly